validation/destination/destination_validation.py

```diff
@@ -1,67 +1,186 @@
-    nearest_dists = []
-    for item in gdf_osm.geometry:
-        nearest_dist = gdf_official.distance(item).min()
-        nearest_dists.append(nearest_dist)
-    gdf_osm['nearest_distance'] = nearest_dists
-    print(gdf_osm['nearest_distance'])
-
-
-min_distance(gdf_osm,gdf_official)
-#Graphing the histogram of distance from gdf_osm to nearest Official points
-ax = gdf_osm['nearest_distance'].hist()
-plt.show()
-
-mean_dist = gdf_osm['nearest_distance'].mean()
-print("mean nearest distance: ",mean_dist)
-
-median_dist = gdf_osm['nearest_distance'].median()
-print("median nearest distance: ", median_dist)
-
-
+import pandas as pd
+
+import osmnx as ox
+
+# configure script
+cities = ['olomouc', 'belfast', 'sao_paulo']
+edge_buffer_dists = [0, 10, 50]
+indicators_filepath = './indicators.csv'
+figure_filepath = './fig/destination-comparison-{city}.png'
+
+def load_data(osm_buffer_gpkg_path, official_dests_filepath, dests_column_name):
+    """
+    Load the city destinations and study boundary.
+
+    Parameters
+    ----------
+    osm_buffer_gpkg_path : str
+        path to the buffered study area geopackage
+    official_dests_filepath : str
+        path to the official destinations shapefile
+    dests_column_name : str
+        name of columns that will be searched for in destination date
+
+    Returns
+    -------
+    study_area, gdf_official_destinations, gdf_osm_destinations : tuple
+    	the polygon composed of square kilometers that is the city's study area, 
+    	the destinations from the official data source,
+    	the destinations sourced from OSM
+    """
+
+    # load the study area boundary as a shapely (multi)polygon
+    gdf_study_area = gpd.read_file(osm_buffer_gpkg_path, layer='urban_study_region')
+    study_area = gdf_study_area['geometry'].iloc[0]
+    print(ox.ts(), 'loaded study area boundary')
+
+    # load the official destinatinos shapefile
+    gdf_official_destinations = gpd.read_file(official_dests_filepath)
+    print(ox.ts(), 'loaded official destinations shapefile')
+
+    # load the osm destinations shapefile
+    gdf_osm = gpd.read_file(osm_buffer_gpkg_path, layer = 'destinations')
+    gdf_osm_destinations = gdf_osm[gdf_osm['dest_name'] == dests_column_name]
+    print(ox.ts(), 'loaded osm destinations shapefile')
+
+	# Project the data to a common crs
+    crs = gdf_study_area.crs
+    if gdf_official_destinations.crs != crs:
+        gdf_official_destinations = gdf_official_destinations.to_crs(crs)
+        print(ox.ts(), 'projected official destinations')
+    if gdf_osm_destinations.crs != crs:
+        gdf_osm_destinations = gdf_osm_destinations.to_crs(crs)
+        print(ox.ts(), 'projected osm destinations')
+    
+    # double-check everything has same CRS, then return
+    assert gdf_study_area.crs == gdf_official_destinations.crs == gdf_osm_destinations.crs
+    return study_area, gdf_official_destinations, gdf_osm_destinations
+
+def total_destination_count(gdf_destinations):
```

Comment:
we can simplify this script by deleting this function, and whenever it is called elsewhere just replace that call with `len(gdf_whatever)` instead